src/gui/transferlistmodel.cpp

```diff
@@ -179,37 +203,179 @@ QVariant TransferListModel::headerData(int section, Qt::Orientation orientation,
+                : QString::fromUtf8(C_INFINITY);
+    };
+
+    const auto amountString = [hideValues](const qint64 value, const qint64 total) -> QString
+    {
+        return ((value == 0) && (total == 0) && hideValues)
+                ? QString {}
+                : QString::number(value) + " (" + QString::number(total) + ')';
+    };
+
+    const auto ratioString = [hideValues](const qreal value) -> QString
+    {
+        if ((value <= 0) && hideValues)
+            return {};
+
+         return ((static_cast<int>(value) == -1) || (value > BitTorrent::TorrentHandle::MAX_RATIO))
+                  ? QString::fromUtf8(C_INFINITY) : Utils::String::fromDouble(value, 2);
+    };
+
+    const auto queuePositionString = [](const qint64 value) -> QString
+    {
+        return (value > 0) ? QString::number(value) : "*";
+    };
+
+    const auto lastActivityString = [hideValues](qint64 value) -> QString
+    {
+        if (hideValues && ((value < 0) || (value >= MAX_ETA)))
+            return QString {};
+
+        // Show '< 1m ago' when elapsed time is 0
+        if (value == 0)
+            value = 1;
+
+        return (value >= 0)
+            ? tr("%1 ago", "e.g.: 1h 20m ago").arg(Utils::Misc::userFriendlyDuration(value))
+            : Utils::Misc::userFriendlyDuration(value);
+    };
+
+    const auto timeElapsedString = [](const qint64 elapsedTime, const qint64 seedingTime) -> QString
+    {
+        if (seedingTime <= 0)
+            return Utils::Misc::userFriendlyDuration(elapsedTime);
+
+        return tr("%1 (seeded for %2)", "e.g. 4m39s (seeded for 3m10s)")
+                .arg(Utils::Misc::userFriendlyDuration(elapsedTime)
+                     , Utils::Misc::userFriendlyDuration(seedingTime));
+    };
+
+    const auto tagsString = [](const QSet<QString> &tags) -> QString
+    {
+        QStringList tagsList = tags.values();
+        tagsList.sort();
+        return tagsList.join(", ");
+    };
+
+    const auto progressString = [](qreal progress) -> QString
+    {
+        progress *= 100;
+        return (static_cast<int>(progress) == 100)
+                ? QString {QLatin1String {"100%"}}
+                : Utils::String::fromDouble(progress, 1) + '%';
+    };
+
+    switch (column) {
+    case TR_NAME:
+        return torrent->name();
+    case TR_QUEUE_POSITION:
+        return queuePositionString(torrent->queuePosition());
+    case TR_SIZE:
+        return unitString(torrent->wantedSize());
+    case TR_PROGRESS:
+        return progressString(torrent->progress());
+    case TR_STATUS:
+        return statusString(torrent->state(), torrent->error());
+    case TR_SEEDS:
+        return amountString(torrent->seedsCount(), torrent->totalSeedsCount());
+    case TR_PEERS:
+        return amountString(torrent->leechsCount(), torrent->totalLeechersCount());
+    case TR_DLSPEED:
+        return unitString(torrent->downloadPayloadRate());
+    case TR_UPSPEED:
+        return unitString(torrent->uploadPayloadRate());
```

Comment:
These 2 should be `return unitString(torrent->uploadPayloadRate(), true);`

Reply:
Thank you. Fixed!